### PR TITLE
feat:support baidu vector db

### LINKED ISSUE #8592: Support baidu vector db
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Dify doesn't support baidu vector db(https://cloud.baidu.com/product/vdb.html) yet, it has rich features and high performance:
- support vector search && bm25 search && hybrid search
- support rich index types, FLAT && HNSW && PQ && PUCK
- higher performance

would you guys be willing to accept a PR if i were to implement this with pymochow?

### 2. Additional context or comments

No response

### 3. Can you help us with this feature?

- [x] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by WayneCao**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Fixes https://github.com/langgenius/dify/issues/8592 ; The goal is to add Baidu VectorDB support. Currently, Baidu VectorDB is not included in vector database support. Baidu Vector Database refers to a database system developed by Baidu that is specifically designed to handle and manage vector data efficiently，it's optimized to handle large-scale vector data with high performance and low latency, enabling advanced applications in areas such as natural language processing, computer vision, and personalized recommendations.

Fixes 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

## Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

 - [ ] Test A
 - [ ] Test B

### CODE REVIEW

File: api/configs/middleware/vdb/baidu_vector_config.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    """
+
+    BAIDU_VECTOR_DB_ENDPOINT: Optional[str] = Field(
+        description="URL of the Baidu Vector Database service (e.g., 'http://vdb.bj.baidubce.com')",
+        default=None,
+    )
+
+    BAIDU_VECTOR_DB_CONNECTION_TIMEOUT_MS: PositiveInt = Field(
+        description="Timeout in milliseconds for Baidu Vector Database operations (default is 30000 milliseconds)",
+        default=30,
```
Comment (bowenliang123) [inline api/configs/middleware/vdb/baidu_vector_config.py]:
```suggestion
        default=30000,
```

File: api/pyproject.toml [hunk 2]
```diff
@@ -190,6 +190,7 @@ zhipuai = "1.0.7"
 # Related transparent dependencies with pinned version
 # required by main implementations
 ############################################################
+pymochow = "1.3.1"
```
Comment (bowenliang123) [inline api/pyproject.toml]: Avoid repeating the dependency crosssing groups. Moreover, it's used for vdb clients, which should not be put in main or indirect dependency group.

File: api/configs/middleware/vdb/baidu_vector_config.py (hunk 1 above)
Comment (WayneCao) [reply api/configs/middleware/vdb/baidu_vector_config.py]: thanks, fixed

File: api/pyproject.toml (hunk 2 above)
Comment (WayneCao) [reply api/pyproject.toml]: thanks, fixed

Comment (bowenliang123) [reply api/pyproject.toml]: LGTM.

**Comment by bowenliang123** [comment]
cc @JohnJyong

**Review by crazywoola — DISMISSED** [review]
LGTM

**Comment by crazywoola** [comment]
Please

**Comment by crazywoola** [comment]
Please resolve the deps

**Comment by WayneCao** [comment]
> Please resolve the deps

resolved

**Comment by crazywoola** [comment]
```
Error: pyproject.toml changed significantly since poetry.lock was last generated. Run `poetry lock [--no-update]` to fix the lock file.
```

**Comment by WayneCao** [comment]
@crazywoola 

> ```
> Error: pyproject.toml changed significantly since poetry.lock was last generated. Run `poetry lock [--no-update]` to fix the lock file.
> ```

sorry, fixed

**Comment by crazywoola** [comment]
Tests

**Comment by WayneCao** [comment]
> Tests are failed.

fixed, i removed baidu vector db from integration_test, Baidu VectorDB not provide opensource version currently.

**Review by crazywoola — APPROVED** [review]

**Pull request merged**